File: smart_salon_app/lib/data/models/appointment.dart
class AppointmentItem {
  final int id;
  final DateTime startTime;
  final DateTime endTime;
  final int status;
  final double estimatedPrice;
  final double depositAmount;
  final bool isRated;
  final int rating;
  final String? comment;
  final String salonName;
  final String artistName;
  final String serviceName;

  AppointmentItem({
    required this.id,
    required this.startTime,
    required this.endTime,
    required this.status,
    this.estimatedPrice = 0,
    this.depositAmount = 0,
    this.isRated = false,
    this.rating = 0,
    this.comment,
    this.salonName = '',
    this.artistName = '',
    this.serviceName = '',
  });

  factory AppointmentItem.fromJson(Map<String, dynamic> json) {
    return AppointmentItem(
      id: json['id'] ?? 0,
      startTime: DateTime.parse(json['startTime']),
      endTime: DateTime.parse(json['endTime']),
      status: json['status'] ?? 0,
      estimatedPrice: (json['estimatedPrice'] ?? 0).toDouble(),
      depositAmount: (json['depositAmount'] ?? 0).toDouble(),
      isRated: json['isRated'] ?? false,
      rating: json['rating'] ?? 0,
      comment: json['comment'],
      salonName: json['salonName'] ?? '',
      artistName: json['artistName'] ?? '',
      serviceName: json['serviceName'] ?? '',
    );
  }

  String get statusText {
    switch (status) {
      case 1: return 'در انتظار';
      case 2: return 'تایید شده';
      case 3: return 'در حال انجام';
      case 4: return 'تمام شده';
      case 5: return 'لغو شده';
      case 6: return 'حضور نیافت';
      default: return 'نامشخص';
    }
  }
}

enum AppointmentStatus {
  pending(1, 'در انتظار'),
  confirmed(2, 'تایید شده'),
  inProgress(3, 'در حال انجام'),
  completed(4, 'تمام شده'),
  cancelled(5, 'لغو شده'),
  noShow(6, 'حضور نیافت');

  final int value;
  final String label;
  const AppointmentStatus(this.value, this.label);
}

File: SmartSalon/SmartSalon/Models/Appointment.cs
namespace SmartSalon.Models
{
    public class Appointment
    {
        public int Id { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public int DurationMinutes { get; set; }
        public decimal EstimatedPrice { get; set; }
        public decimal? FinalPrice { get; set; }
        public decimal DepositAmount { get; set; }
        public AppointmentStatus Status { get; set; } = AppointmentStatus.Pending;
        public string? Notes { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // این رزرو برای کدام مشتری است؟
        public string ClientId { get; set; } = string.Empty;
        public ApplicationUser? Client { get; set; }

        // این رزرو برای کدام هنرمند است؟
        public int ArtistId { get; set; }
        public Artist? Artist { get; set; }

        // این رزرو در کدام سالن است؟
        public int SalonId { get; set; }
        public Salon? Salon { get; set; }

        // این رزرو برای کدام خدمت است؟
        public int ServiceId { get; set; }
        public SalonService? Service { get; set; }

        public bool IsRated { get; set; } = false;
        public int Rating { get; set; } = 0;
        public string? Comment { get; set; }
        public bool ReminderSent { get; set; } = false;
    }

    public enum AppointmentStatus
    {
        Pending = 1,  // در انتظار تایید
        Confirmed = 2,  // تایید شده
        InProgress = 3,  // در حال انجام
        Completed = 4,  // تمام شده
        Cancelled = 5,  // لغو شده
        NoShow = 6   // مشتری نیامد
    }
}